var/spack/repos/builtin/packages/tar/package.py

```diff
@@ -27,6 +29,12 @@ class Tar(AutotoolsPackage, GNUMirrorPackage):
     patch('argp-pgi.patch',   when='@:1.29')
     patch('gnutar-configure-xattrs.patch', when='@1.28')
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('--version', output=str)
+        match = re.search(r'tar \(GNU tar\) (\d+(?:\.\d+)*)', output)
```

Comment:
I would simplify this like so:
```suggestion
        match = re.search(r'tar \(GNU tar\) (\S+)', output)
```